arm_controller:
  follow_joint_trajectory:
    type: "robot_controllers/FollowJointTrajectoryController"
    joints:
      - shoulder_roll_joint
      - shoulder_pitch_joint
      - shoulder_yaw_joint
      - elbow_pitch_joint
      - elbow_yaw_joint
      - wrist_pitch_joint
      - wrist_roll_joint
  gravity_compensation:
    type: "robot_controllers/GravityCompensation"
    root: "top_plate"
    tip: "virtual_endeffector"
    autostart: true
  
gripper_controller:
  gripper_action:
    type: "robot_controllers/ParallelGripperController"
    centering:
      p: 1000.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0

controller_gains:
    body:
        linear:
            p: 0.70
            i: 0.01
            d: 0.1
        angular:
            p: -1.00
            i: 0.01
            d: 0.1

gazebo:
  default_controllers:
    - "arm_controller/follow_joint_trajectory"
    - "arm_controller/gravity_compensation"
    - "gripper_controller/gripper_action"
  swivel_hubcap_joint:
    position:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
    velocity:
      p: 8.85
      d: 0.0
      i: 0.5
      i_clamp: 6.0
  base_swivel_joint:
    position:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
    velocity:
      p: 8.85
      d: 0.0
      i: 0.5
      i_clamp: 6.0
  l_wheel_joint:
    position:
      p: 0.70
      d: 0.01
      i: 0.1
      i_clamp: 6.0
    velocity:
      p: 0.70
      d: 0.01
      i: 0.1
      i_clamp: 6.0
  r_wheel_joint:
    position:
      p: 0.70
      d: 0.01
      i: 0.1
      i_clamp: 6.0
    velocity:
      p: 0.70
      d: 0.01
      i: 0.1
      i_clamp: 6.0
  shoulder_roll_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  shoulder_pitch_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  shoulder_yaw_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  elbow_pitch_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  elbow_yaw_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  wrist_pitch_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  wrist_roll_joint:
    position:
      p: 10.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 10.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  l_gripper_finger_joint:
    position:
      p: 5000.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  r_gripper_finger_joint:
    position:
      p: 5000.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
